src/databricks/labs/ucx/source_code/python_libraries.py

```diff
@@ -45,18 +47,41 @@ def _temporary_virtual_environment(self):
 
     def _install_library(self, path_lookup: PathLookup, library: Path) -> list[DependencyProblem]:
         """Pip install library and augment path look-up to resolve the library at import"""
-        venv = self._temporary_virtual_environment
-        path_lookup.append_path(venv)
-        # resolve relative pip installs from notebooks: %pip install ../../foo.whl
+        path_lookup.append_path(self._temporary_virtual_environment)
+
+        # Resolve relative pip installs from notebooks: %pip install ../../foo.whl
         maybe_library = path_lookup.resolve(library)
         if maybe_library is not None:
             library = maybe_library
-        return_code, stdout, stderr = self._runner(f"pip install {library} -t {venv}")
+
+        if library.suffix == ".egg":
+            return self._install_egg(library)
+        return self._install_pip(library)
+
+    def _install_pip(self, library: Path) -> list[DependencyProblem]:
+        return_code, stdout, stderr = self._runner(f"pip install {library} -t {self._temporary_virtual_environment}")
         logger.debug(f"pip output:\n{stdout}\n{stderr}")
         if return_code != 0:
             problem = DependencyProblem("library-install-failed", f"Failed to install {library}: {stderr}")
             return [problem]
         return []
 
+    def _install_egg(self, library: Path) -> list[DependencyProblem]:
+        verbosity = "--verbose" if is_in_debug() else "--quiet"
+        easy_install_arguments = [
+            "easy_install",
+            verbosity,
+            "--always-unzip",
+            "--install-dir",
+            self._temporary_virtual_environment.as_posix(),
+            library.as_posix(),
+        ]
+        try:
+            setup(script_args=easy_install_arguments)
+        except SystemExit as e:
```

Comment:
```suggestion
            from setuptools import setup
            setup(script_args=easy_install_arguments)
        except SystemExit, ImportError as e:
```
my professional gut feeling suggests that we guard an import to this ancient library with import error to protect us from cases, where this thing is not available.